### PR TITLE
Enforce stricter validation of Linux process maps file

### PR DESCRIPTION
**Pull request by jiangliu**
Not trust content from /proc/pid/maps and enforce stricter validation.

### CODE REVIEW

File: src/linux_maps.rs [hunk 1]
```diff
@@ -51,49 +51,96 @@ pub fn get_process_maps(pid: Pid) -> std::io::Result<Vec<MapRange>> {
     // Parses /proc/PID/maps into a Vec<MapRange>
     let maps_file = format!("/proc/{}/maps", pid);
     let mut file = File::open(maps_file)?;
+
+    // Check that the file is not too big
+    let metadata = file.metadata()?;
+    if metadata.len() > 0x1000000 {
```
Comment (acj) [inline src/linux_maps.rs]: Can you clarify what "too big" means in this case? Is this based on an actual limit in the kernel?

**Review by acj — COMMENTED** [review]
Thanks for working on this. Just one question, and then I think we can merge.

File: src/linux_maps.rs (hunk 1 above)
Comment (jiangliu) [reply src/linux_maps.rs]: It's defensive and arbitrary. If feel risky, I can enlarge it limitation or remove it.

Comment (jiangliu) [reply src/linux_maps.rs]: The default configuration for Linux is 65530 vmas, each vma make take up to 4k+ bytes at most. So seems we need to enlarge the constraint to 0x10000000 at least.

Comment (jiangliu) [reply src/linux_maps.rs]: updated

**Comment by acj** [comment]
Published in v0.4.0.

**Pull request merged**